pkg/simulate/pushsync/pushsync_test.go

```diff
@@ -0,0 +1,91 @@
+package pushsync_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/ethersphere/beekeeper/pkg/simulate/pushsync"
+)
+
+func TestGetIPFromUnderlays(t *testing.T) {
+
+	ips := []string{
+		"/ip4/127.0.0.1/udp/9090/quic",
+		"/ip6/::1/tcp/3217",
+		"/ip4/10.0.0.0/tcp/80/http/baz.jpg",
+	}
+	ip := pushsync.GetIPFromUnderlays(ips)
+
+	if ip != "10.0.0.0" {
+		t.Fatalf("want %s got %s", ip, "10.0.0.0")
+	}
+}
+
+func TestTobuckets(t *testing.T) {
+
+	for _, tc := range []struct {
+		name     string
+		base     []string
+		buckets  [][]string
+		leftover []string
+		start    float64
+		end      float64
+		step     float64
+	}{
+		{
+			name:     "0.1-0.8-0.1",
+			base:     []string{"1", "2", "3", "4", "5", "6", "7", "8", "9", "10"},
+			buckets:  [][]string{{"1"}, {"2"}, {"3"}, {"4"}, {"5"}, {"6"}, {"7"}, {"8"}},
+			leftover: []string{"9", "10"},
```

Comment:
the leftover part can perhaps be simplified a bit. usually you would test some scenario:
- 90% nodes running correctly and 10% misbehaving (the tests would always be conducted on the !misbehaving nodes)
- 80% ok, 20% misbehaving
- etc...

so maybe buckets could just reflect that and then the whole leftover part can be dropped. so each bucket could reflect the whole dataset instead of having to know the state of previous buckets (i.e. the 80% bucket will contain 80% of the addresses, instead of having to step over the previous bins. _however_ i do understand that it simplifies the management of the misbehaving eth nodes (so that you don't need to do the same operation twice). still, the readability can probably be improved here... perhaps you have some better suggestion?

Reply:
also, just another thought on this is that if you count from 100% down, you could probably remove leftovers.
since you know the lower bound of the percentage you're aiming at, that would always be your `leftovers`, and the buckets would just reflect everything between that and 100% just as they do today

Reply:
So I removed the leftover return value, and now the last bucket is the "leftover" or "out of range" elements.

Reply:
> also, just another thought on this is that if you count from 100% down, you could probably remove leftovers.
> since you know the lower bound of the percentage you're aiming at, that would always be your `leftovers`, and the buckets would just reflect everything between that and 100% just as they do today

Hopefully, this was what you meant :)